fancy-weather/scripts/script.js

```diff
@@ -0,0 +1,447 @@
+  "Января",
+  "Февраля",
+  "Марта",
+  "Апреля",
+  "Мая",
+  "Июня",
+  "Июля",
+  "Августа",
+  "Сентября",
+  "Октября",
+  "Ноября",
+  "Декабря",
+];
+
+const mapOptions = {
+  enableHighAccuracy: true,
+  timeout: 5000,
+  maximumAge: 0,
+};
+
+async function setBackground() {
+  try {
+    const res = await fetch(urlImage);
+    const data = await res.json();
+    document.body.style.backgroundImage = `url(${data.urls.regular})`;
+  } catch (e) {
+    setBackgroundFromLibrary();
+  }
+}
+
+function setBackgroundFromLibrary() {
+  const num = Math.floor(Math.random() * imageLinksLibrary.length);
+  document.body.style.backgroundImage = `url(${imageLinksLibrary[num]})`;
+}
+
+async function getCoordsFromNavigator(position) {
+  const crd = await position.coords;
+  latitude = parseFloat(crd.latitude).toPrecision(4);
+  longitude = parseFloat(crd.longitude).toPrecision(4);
+  getMap();
+  await getWeatherByCoord();
+  getForecast();
+}
+
+async function getCoordsFromNavigatorError(err) {
+  console.warn(`ERROR(${err.code}): ${err.message}`);
+  await getPlaceByIp();
+  getWeatherByCity();
+  getForecast();
+}
+
+async function getPlaceByIp() {
+  try {
+    const resp = await fetch(urlIp);
+    const data = await resp.json();
+    city = data.city;
+    countryCode = data.country;
+    const loc = data.loc.split(",");
+    latitude = loc[0];
+    longitude = loc[1];
+    getMap();
+    getWeatherByCity();
+    getForecast();
+  } catch {
+    console.log("Error, do not get your IP");
+    getMap();
+  }
+}
+
+function timeInsert() {
+  const timeNow = new Date(Date.now() + timeZone * milisecondsInOneSecond);
+  const month = timeNow.getUTCMonth();
+  const day = timeNow.getUTCDay();
+  const date = timeNow.getUTCDate();
+  const hour = timeNow.getUTCHours();
+  const minutes = timeNow.getUTCMinutes();
+  const seconds = timeNow.getUTCSeconds();
+  if (lang === "en") {
+    time.textContent = `${dayOfWeekShortEn[day]} ${date} ${monthTitleEn[month]} ${hour.toString().padStart(2, "0")}:${minutes.toString().padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+    dayOfWeek.forEach((item, i) => {
+      item.textContent = dayOfWeekEn[(i + 1 + day) % daysInWeek];
+    });
+  } else {
+    time.textContent = `${dayOfWeekShortRu[day]} ${date} ${monthTitleRu[month]} ${hour.toString().padStart(2, "0")}:${minutes.toString().padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+    dayOfWeek.forEach((item, i) => {
+      item.textContent = dayOfWeekRu[(i + 1 + day) % daysInWeek];
+    });
+  }
+  setTimeout(timeInsert, 1000);
+}
+
+function getLang() {
+  if (localStorage.getItem("lang") === null) {
+    localStorage.setItem("lang", "en");
+    lang = en;
+  } else {
+    lang = localStorage.getItem("lang");
+  }
+}
+
+function getTempScale() {
+  if (localStorage.getItem("tempFlag") === null) {
+    localStorage.setItem("tempFlag", "c");
+    tempFlag = "c";
+  } else {
+    tempFlag = localStorage.getItem("tempFlag");
```

Comment:
Almost same functionality as getLang
Common parts could be moved to helper functions